### PR TITLE
Add top nav bar, move edit button onto it, prototype session selector

### PR DESCRIPTION
**Pull request by ngzhian**
This PR adds a top navigation bar, which contains the session selector, the name of our app, and a control for switching between modes. Please see attached screenshot to check out how it looks.

The session selector is currently a plain text button, and clicking on it shows a `TableViewController` (containing dummy data for sessions right now) in a `popover` style. Tapping on the sessions in the table does nothing as of now.

The mode switch is implemented as a `UISegmentedControl`, overridden with a custom tint colour to match the rest of the interface. It behaves the same way as the ugly red `EDIT` button previously, i.e. shows/hides the side pane.

No changes are made to the side pane.

I shuffled some files around too. But all UI related so shouldn't conflict.

simulator screen shot 20 mar 2017 10 24 49 pm

### CODE REVIEW

File: MIDIchlorians/MIDIchlorians/ViewController.swift [hunk 1]
```diff
@@ -46,67 +46,90 @@ class ViewController: UIViewController {
...
         UITableView.appearance().backgroundColor = Config.BackgroundColor
         UITableViewCell.appearance().backgroundColor = Config.BackgroundColor
     }
 
+    // Sets up the top navigation.
+    // The top navigation has controls to show the session table, so we set that up here as well.
+    private func setUpTopNav() {
+        sessionTableViewController = SessionTableViewController(style: .plain)
+        sessionNavigationController = UINavigationController(rootViewController: sessionTableViewController)
+        topNavigationBar = TopNavigationBar(frame: CGRect(x: 0, y: 0, width: self.view.frame.width, height: 60))
```
Comment (anand-sundaram) [inline MIDIchlorians/MIDIchlorians/ViewController.swift:92]: remove magic numbers?

File: MIDIchlorians/MIDIchlorians/ViewController.swift [hunk 2]
```diff
@@ -46,67 +46,90 @@ class ViewController: UIViewController {
...
+        // configure styles and anchor of popover
+        let popoverPresentationController = sessionNavigationController.popoverPresentationController
+        popoverPresentationController?.permittedArrowDirections = [.up]
+        popoverPresentationController?.barButtonItem = sender
+        popoverPresentationController?.backgroundColor = Config.BackgroundColor
+    }
+
+    private func fixGridDimensions() {
         // fix the width of the button collection view
         let totalWidth = self.view.frame.width - 20 - 20 // padding left and right
```
Comment (anand-sundaram) [inline MIDIchlorians/MIDIchlorians/ViewController.swift:134]: remove magic number?

**Comment by anand-sundaram** [comment]
LGTM except for magic numbers (those can be removed later on as well, along with all the other hardcoded stuff :))

**Review by anand-sundaram — COMMENTED** [review]
LGTM except for magic numbers (those can be removed later on as well, along with all the other hardcoded stuff :))

**Comment by ngzhian** [comment]
Will remove magic numbers in a next PR, thinking of replacing it with constraints.

**Pull request merged**